master/internal/resourcemanagers/kubernetes_resource_manager.go

```diff
@@ -253,31 +290,64 @@ func (k *kubernetesResourceManager) receiveJobQueueMsg(ctx *actor.Context) error
 					group.qPosition = msg.QPosition
 					ctx.Tell(req.Group, sproto.SetGroupOrder{
 						QPosition: msg.QPosition,
-						Handler:   ctx.Self(),
 					})
 				}
 				if *msg.Priority > 0 {
 					group.priority = msg.Priority
 					ctx.Tell(req.Group, sproto.SetGroupPriority{
 						Priority: msg.Priority,
-						Handler:  ctx.Self(),
-					})
-				}
-				if msg.Weight > 0 {
-					group.weight = msg.Weight
-					ctx.Tell(req.Group, sproto.SetGroupWeight{
-						Weight:  msg.Weight,
-						Handler: ctx.Self(),
 					})
 				}
+				// do nothing if message is setting the weight
 			}
 		}
+	//case job.GetJobSummary:
+	//	resp := k.getJobSummary(msg.JobID)
+	//	ctx.Respond(resp)
+	case job.GetJobQStats:
+		ctx.Respond(*k.getJobQStats())
 	default:
 		return actor.ErrUnexpectedMessage(ctx)
 	}
 	return nil
 }
 
+//func (k *kubernetesResourceManager) getJobSummary(jobID model.JobID) *jobv1.JobSummary {
+//	var requests []*sproto.AllocateRequest
+//	for it := k.reqList.iterator(); it.next(); {
+//		requests = append(requests, it.value())
+//	}
+//	jobs := mergeToJobs(requests, k.groups, kubernetesScheduler)
+//	for _, job := range jobs {
+//		if job.JobId == jobID.String() {
+//			return job.Summary
+//		}
+//	}
+//	return nil
+//}
+
+func (k *kubernetesResourceManager) getJobQStats() *jobv1.QueueStats {
```

Comment:
inside the k8s resource manager, the functions used and arguments are a bit different, so I more or less redefined them here. I think consolidating them might be the right move, and I'll either add that in a different PR or update this PR in the next couple of days.

Reply:
maybe leave an inline todo as well so this doesn't get lost?